xtask/src/swift.rs

```diff
@@ -66,7 +69,68 @@ impl SwiftArgs {
     }
 }
 
+/// A specific build target supported by the SDK.
+struct Target {
+    triple: &'static str,
+    platform: Platform,
+    description: &'static str,
+}
+
+/// The platform for which a particular target can run on.
+#[derive(Hash, PartialEq, Eq, Clone)]
+enum Platform {
+    Macos,
+    Ios,
+    IosSimulator,
+}
+
+impl Platform {
+    /// The human-readable name of the platform.
+    fn as_str(&self) -> &str {
+        match self {
+            Platform::Macos => "macOS",
+            Platform::Ios => "iOS",
+            Platform::IosSimulator => "iOS Simulator",
+        }
+    }
+
+    /// The name of the library for the platform once all architectures are
+    /// lipo'd together.
+    fn lib_name(&self) -> &str {
+        match self {
+            Platform::Macos => "libmatrix_sdk_ffi_macos.a",
+            Platform::Ios => "libmatrix_sdk_ffi_ios.a",
+            Platform::IosSimulator => "libmatrix_sdk_ffi_iossimulator.a",
+        }
+    }
+}
+
+/// The base name of the FFI library.
 const FFI_LIBRARY_NAME: &str = "libmatrix_sdk_ffi.a";
+/// The list of targets supported by the SDK.
+const TARGETS: &[Target] = &[
```

Comment:
Can we please define Catalyst as well while we're at it?

Reply:
Will do in a follow-up PR